src/pages/SingleProductPage/components/ProductContainer/ProductContainer.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import '../../SingleProductPage.css';
+import axios from 'axios';
+import { realMadrid } from '../../../../assets';
+function ProductContainer() {
+  const params = useParams();
+  const [data, setData] = useState({ imgSrc: '', title: '', price: '' });
+  useEffect(() => {
+    async function getProducts() {
+      const response = await axios.get('/api/products');
+
+      setData(
+        ...response.data.products.filter(item => item.id === params.productId)
```

Comment:
Instead of **filter** you should use **find** as you need just a single product over here and filter returns an array while **find** will return that specific product and then you might also not need to spread it.